Content.Server/Lightning/LightningSystem.cs

```diff
@@ -0,0 +1,149 @@
+    [Dependency] private readonly BeamSystem _beam = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LightningComponent, StartCollideEvent>(OnCollide);
+        SubscribeLocalEvent<LightningComponent, ComponentRemove>(OnRemove);
+    }
+
+    private void OnRemove(EntityUid uid, LightningComponent component, ComponentRemove args)
+    {
+        if (!TryComp<BeamComponent>(uid, out var lightningBeam)
+            || lightningBeam.VirtualBeamController == null
+            || !TryComp<BeamComponent>(lightningBeam.VirtualBeamController, out var beamController))
+        {
+            return;
+        }
+
+        beamController.CreatedBeams.Remove(uid);
+    }
+
+    private void OnCollide(EntityUid uid, LightningComponent component, StartCollideEvent args)
+    {
+        if (!TryComp<BeamComponent>(uid, out var lightningBeam)
+            || lightningBeam.VirtualBeamController == null
+            || !TryComp<BeamComponent>(lightningBeam.VirtualBeamController, out var beamController))
+        {
+            return;
+        }
+
+        if (component.CanArc)
+        {
+            while (beamController.CreatedBeams.Count < component.MaxTotalArcs)
+            {
+                Arc(component, args.OtherFixture.Body.Owner, lightningBeam.VirtualBeamController.Value);
+
+                var spriteState = LightningRandomizer();
+
+                component.ArcTargets.Add(args.OtherFixture.Body.Owner);
+                component.ArcTargets.Add(component.ArcTarget);
+
+                _beam.TryCreateBeam(args.OtherFixture.Body.Owner, component.ArcTarget, component.LightningPrototype, spriteState, controller: lightningBeam.VirtualBeamController.Value);
+
+                //Break from this loop so other created bolts can collide and arc
+                break;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Fires lightning from user to target
+    /// </summary>
+    /// <param name="user">Where the lightning fires from</param>
+    /// <param name="target">Where the lightning fires to</param>
+    /// <param name="lightningPrototype">The prototype for the lightning to be created</param>
+    public void ShootLightning(EntityUid user, EntityUid target, string lightningPrototype = "Lightning")
+    {
+        var spriteState = LightningRandomizer();
+        _beam.TryCreateBeam(user, target, lightningPrototype, spriteState);
+    }
+
+    /// <summary>
+    /// Picks a random sprite state for the lightning.
+    /// </summary>
+    /// <returns>Returns a sprite state of lightning_ + the chosen random number</returns>
+    private string LightningRandomizer()
+    {
+        //When the lightning is made with TryCreateBeam, spawns random sprites for each beam to make it look nicer.
+        var spriteStateNumber = _random.Next(1, 12);
+        return ("lightning_" + spriteStateNumber);
+    }
```

Comment:
again this needs to be done clientside

Reply:
This isn't setting anything sprite wise, just gets a value ready.